projects/storefrontlib/src/cms-components/myaccount/update-email/update-email.component.ts

```diff
@@ -1,69 +1,23 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
-import {
-  AuthService,
-  GlobalMessageService,
-  GlobalMessageType,
-  RoutingService,
-  UserService,
-} from '@spartacus/core';
-import { Observable, Subscription } from 'rxjs';
+import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { UpdateEmailService } from './update-email.service';
 
 @Component({
   selector: 'cx-update-email',
   templateUrl: './update-email.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class UpdateEmailComponent implements OnInit, OnDestroy {
-  constructor(
-    private routingService: RoutingService,
-    private globalMessageService: GlobalMessageService,
-    private userService: UserService,
-    private authService: AuthService
-  ) {}
+export class UpdateEmailComponent implements OnDestroy {
+  constructor(protected service: UpdateEmailService) {}
 
-  private subscription = new Subscription();
-  private newUid: string;
-  isLoading$: Observable<boolean>;
-
-  ngOnInit() {
-    this.userService.resetUpdateEmailResultState();
-    this.subscription.add(
-      this.userService
-        .getUpdateEmailResultSuccess()
-        .subscribe((success) => this.onSuccess(success))
-    );
-    this.isLoading$ = this.userService.getUpdateEmailResultLoading();
-  }
-
-  onCancel(): void {
-    this.routingService.go({ cxRoute: 'home' });
-  }
-
-  onSubmit({ newUid, password }: { newUid: string; password: string }): void {
-    this.newUid = newUid;
-    this.userService.updateEmail(password, newUid);
-  }
-
-  async onSuccess(success: boolean): Promise<void> {
-    if (success) {
-      this.globalMessageService.add(
-        {
-          key: 'updateEmailForm.emailUpdateSuccess',
-          params: { newUid: this.newUid },
-        },
-        GlobalMessageType.MSG_TYPE_CONFIRMATION
-      );
-      // TODO(#9638): Use logout route when it will support passing redirect url
-      await this.authService.coreLogout();
-      this.routingService.go({ cxRoute: 'login' }, null, {
-        state: {
-          newUid: this.newUid,
-        },
-      });
-    }
-  }
+  form: FormGroup = this.service.form;
+  isUpdating$: Observable<boolean> = this.service.isUpdating$;
```

Comment:
nitpick: what about leaving out the typing here? It's implicit from the service method signature, and both in controller and view logic my IDE is fully aware of the type. Or any arguments against it?